pkg/operator/podsecurityreadinesscontroller/conditions.go

```diff
@@ -0,0 +1,89 @@
+package podsecurityreadinesscontroller
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const (
+	ViolationReason = "PodSecurityReadinessController"
+	ExpectedReason  = "AsExpected"
+
+	PodSecurityCustomerType     = "PodSecurityCustomerEvaluationConditionsDetected"
+	PodSecurityOpenshiftType    = "PodSecurityOpenshiftEvaluationConditionsDetected"
+	PodSecurityRunLevelZeroType = "PodSecurityRunLevelZeroEvaluationConditionsDetected"
+)
+
+var (
+	// run-level zero namespaces, shouldn't avoid openshift namespaces
+	runLevelZeroNamespaces = map[string]struct{}{
+		"default":     empty,
+		"kube-system": empty,
+		"kube-public": empty,
+	}
+	empty = struct{}{}
+)
+
+type podSecurityOperatorConditions struct {
+	violatingOpenShiftNamespaces    []string
+	violatingRunLevelZeroNamespaces []string
+	violatingCustomerNamespaces     []string
+}
+
+func (c *podSecurityOperatorConditions) addViolation(name string) {
+	if isRunLevelZero(name) {
+		c.violatingRunLevelZeroNamespaces = append(c.violatingRunLevelZeroNamespaces, name)
+		return
+	}
+
+	isOpenShift := strings.HasPrefix(name, "openshift")
+	if isOpenShift {
+		c.violatingOpenShiftNamespaces = append(c.violatingOpenShiftNamespaces, name)
+		return
+	}
+
+	c.violatingCustomerNamespaces = append(c.violatingCustomerNamespaces, name)
+}
+
+func isRunLevelZero(name string) bool {
+	_, isRunLevelZero := runLevelZeroNamespaces[name]
+	return isRunLevelZero
+
+}
+
+func makeCondition(conditionType string, namespaces []string) operatorv1.OperatorCondition {
+	if len(namespaces) > 0 {
+		sort.Strings(namespaces)
+		return operatorv1.OperatorCondition{
+			Type:               conditionType,
+			Status:             operatorv1.ConditionTrue,
+			LastTransitionTime: metav1.Now(),
+			Reason:             ViolationReason,
+			Message: fmt.Sprintf(
+				"Violations detected in namespaces: %s",
```

Comment:
`%v` with the slice might work better when the condition messages get concatenated